backend_new/prisma/seed-helpers/ami-chart-factory.ts

```diff
@@ -0,0 +1,22 @@
+import { Prisma } from '@prisma/client';
+
+export const amiChartFactory = (
+  i: number,
+  jurisdictionId: string,
+): Prisma.AmiChartCreateInput => ({
+  name: `name: ${i}`,
+  items: amiChartItemsFactory(i),
+  jurisdictions: {
+    connect: {
+      id: jurisdictionId,
+    },
+  },
+});
+
+const amiChartItemsFactory = (numberToCreate: number) => {
+  return JSON.stringify(
```

Comment:
It looks like this is returning `[null,null,null,null,null,null,null,null,null,null]` when `db:setup` is run. Unfortunately:

> callbackFn is invoked only for array indexes which have assigned values. It is not invoked for empty slots in [sparse arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Indexed_collections#sparse_arrays).

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map#description, so you'll have to do something else.

It looks like the return type of `amiChartItemsFactory` could be `Prisma.JsonArray`, which also allows you to get rid of the JSON.stringify.

Something like this should work:
```
const amiChartItemsFactory = (numberToCreate: number): Prisma.JsonArray =>
  new Array(numberToCreate).fill(undefined).map((_, index) => ({
    percentOfAmi: index,
    householdSize: index,
    income: index,
  }));
```

Reply:
ooo, I think I'd prefer this over what I suggested above:
```
const amiChartItemsFactory = (numberToCreate: number): Prisma.JsonArray =>
  [...Array(numberToCreate)].map((_, index) => ({
    percentOfAmi: index,
    householdSize: index,
    income: index,
  }));
```